gno.land/pkg/integration/testing_integration.go

```diff
@@ -369,6 +422,118 @@ func createAccount(env envSetter, kb keys.Keybase, accountName string) (gnoland.
 
 	return gnoland.Balance{
 		Address: address,
-		Amount:  std.Coins{std.NewCoin("ugnot", 1000000000000000000)},
+		Amount:  std.Coins{std.NewCoin("ugnot", 10e6)},
 	}, nil
 }
+
+type pkgsLoader map[string]gnomod.Pkg
+
+func (pkgs pkgsLoader) list() gnomod.PkgList {
+	list := make([]gnomod.Pkg, 0, len(pkgs))
+	for _, pkg := range pkgs {
+		list = append(list, pkg)
+	}
+	return list
+}
+
+func (pkgs pkgsLoader) Txs(creator bft.Address, fee std.Fee, deposit std.Coins) ([]std.Tx, error) {
+	pkgslist, err := pkgs.list().Sort()
+	if err != nil {
+		return nil, fmt.Errorf("uanble to sort packages: %w", err)
+	}
+
+	txs := make([]std.Tx, len(pkgslist))
+	for i, pkg := range pkgslist {
+		tx, err := gnoland.LoadPackage(pkg, creator, fee, deposit)
+		if err != nil {
+			return nil, fmt.Errorf("unable to load pkg %q: %w", pkg.Name, err)
+		}
+		txs[i] = tx
+	}
+
+	return txs, nil
+}
+
+func (pkgs pkgsLoader) loadPackagesFromDir(path string) error {
+	// list all packages from target path
+	pkgslist, err := gnomod.ListPkgs(path)
+	if err != nil {
+		return fmt.Errorf("listing gno packages: %w", err)
+	}
+
+	for _, pkg := range pkgslist {
+		if _, ok := pkgs[pkg.Name]; !ok {
+			pkgs[pkg.Name] = pkg
+		}
+	}
+
+	return nil
+}
+
+func (pkgs pkgsLoader) loadPackages(modroot string, path, name string) error {
+	if path == "" {
+		return fmt.Errorf("no path specified for package")
+	}
+
+	pkg := gnomod.Pkg{
+		Dir:  path,
+		Name: name,
+	}
+
+	if pkg.Name == "" {
+		// try to load `gno.mod` informations
+		gnoModPath := filepath.Join(pkg.Dir, "gno.mod")
+		data, err := os.ReadFile(gnoModPath)
+		if os.IsNotExist(err) {
+			return fmt.Errorf("no name specified /gnomod doesn't exist (%q)", gnoModPath)
+		}
+
+		if err != nil {
+			return fmt.Errorf("unable to load gno.mod: %w", err)
+		}
+
+		gnoMod, err := gnomod.Parse(gnoModPath, data)
+		if err != nil {
+			return fmt.Errorf("%q parse gnomod error: %w", gnoModPath, err)
+		}
+
+		gnoMod.Sanitize()
+		if err := gnoMod.Validate(); err != nil {
+			return fmt.Errorf("validate %q error: %w", gnoModPath, err)
+		}
+
+		// override pkg info with mod infos
+		pkg.Name = gnoMod.Module.Mod.Path
+		pkg.Draft = gnoMod.Draft
+		for _, req := range gnoMod.Require {
+			pkg.Requires = append(pkg.Requires, req.Mod.Path)
+		}
+	}
+
+	if pkg.Draft {
+		return nil // skip draft package
+	}
+
+	// if no require pkgs are declared, use previous one to keep load order
+	// XXX: find a better way to achieve this
+	if !strings.HasPrefix(pkg.Dir, modroot) && len(pkg.Requires) == 0 {
+		pkg.Requires = make([]string, 0, len(pkgs))
+		for name := range pkgs {
+			pkg.Requires = append(pkg.Requires, name)
+		}
+	}
+
+	if _, ok := pkgs[pkg.Name]; ok {
+		return nil // we already know this pkg
+	}
+	pkgs[pkg.Name] = pkg
+
+	for _, pkgpath := range pkg.Requires {
+		pkgpath := filepath.Join(modroot, pkgpath)
+		if err := pkgs.loadPackages(modroot, pkgpath, ""); err != nil {
+			return fmt.Errorf("require %q: %w", pkgpath, err)
+		}
```

Comment:
I'm not a huge fan of recursive processing like this, especially considering this method relies heavily on the state of the object its tied to to execute 🙁 

We should think about having `loadPackages` be an agnostic method that just returns the packages it digs up, and takes in an initial state.

Please resolve this comment even if you don't swap out the implementation, I just wanted to mention this

Reply:
@zivkovicmilos i've made some updates to improve clarity on this method and its related structure. Please tell me what you think. 

I've transitioned from a recursive to an iterative logic, the only downside with this change is the loss of the natural dependency chain in error reporting. 

for the agnostic part, I prefer to maintain the stateful implementation, as this structure is intended for internal use only and this approach simplifies the logic, especially when making multiple calls to this method to aggregate packages within the state of the structure.

Reply:
Looking good 😎